server/src/test/scala/org/http4s/server/middleware/authentication/AuthMiddlewareSpec.scala

```diff
@@ -0,0 +1,48 @@
+package org.http4s.server.middleware.authentication
+
+import cats.data.Kleisli
+import fs2.Task
+import org.http4s.server.AuthMiddleware
+import org.http4s._
+import org.http4s.dsl._
+
+class AuthMiddlewareSpec extends Http4sSpec {
+
+  "AuthMiddleware" should {
+    "work" in {
```

Comment:
I think this should be broken into individual tests, "work" is not specific as to what is occurring were we to see a failure in this test. 

The middle section can be created into a generator, if there is no good way for a property test.